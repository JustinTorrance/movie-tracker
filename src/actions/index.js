export const loadMovies = (movies) => ({
  type: 'LOAD_MOVIES',
  movies
})

export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const loadFavorites = (favorites) => ({
  type: 'LOAD_FAVORITES',
  favorites
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  bool
})