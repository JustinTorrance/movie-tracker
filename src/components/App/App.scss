@import '../Login/login';

$color-blue: #64d5f8; 
$title-font: 'Squada One', cursive;

.movie-tracker {
  color: $color-blue;
  font-family: $title-font;
  font-size: 80px;
  margin: 10px;
}