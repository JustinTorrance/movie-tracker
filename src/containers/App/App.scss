@import url('https://fonts.googleapis.com/css?family=Raleway:300,400,500');

$color-blue: #64d5f8; 
$title-font: 'Squada One', cursive;
$regular-font: 'Raleway', sans-serif;

@import '../Login/login';
@import '../../components/Signup/signup';
@import '../../components/MovieDisplay/movieDisplay';
@import '../../components/MovieCard/movieCard';
@import '../../components/Loading/Loading';

body {
  background-image: url('../../utils/images/app_background.png');
  background-size: cover;
  background-position: fixed;
  background-blend-mode: overlay;
  background-color: rgba(0, 0, 0, 0.822)
}
