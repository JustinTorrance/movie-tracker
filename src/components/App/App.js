import React, { Component } from 'react'
import MovieDisplay from '../MovieDisplay/MovieDisplay'
import { fetchData } from '../../utils/apiCalls'
import { connect } from 'react-redux'
import { loadMovies } from '../../actions/index'
import './App.scss'
import { Route, Switch } from 'react-router-dom'
import Login from '../Login/Login'

class App extends Component {
  constructor() {
    super()
  }

  async componentDidMount() {
    const movies = await fetchData('https://api.themoviedb.org/3/movie/popular?api_key=da90047b6c1d3526d4b04666a1b64a0d&language=en-US&page=1&region=US')
    this.props.loadMovies(movies)
  }

  render() {
    return (
      <div className="App">
        <h1 className="movie-tracker">movie tracker</h1>
        <Switch>
          <Route 
            exact
            path='/'
            component={MovieDisplay}
          />
          <Route
            path='/login'
            component={Login}
          />
        </Switch>
      </div>

    )
  }
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

const mapDispatchToProps = (dispatch) => ({
  loadMovies: (movies) => dispatch(loadMovies(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
