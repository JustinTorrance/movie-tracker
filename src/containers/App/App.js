import React, { Component } from 'react'
import MovieDisplay from '../../components/MovieDisplay/MovieDisplay'
import { fetchData } from '../../utils/apiCalls'
import { connect } from 'react-redux'
import { loadMovies } from '../../actions/index'
import './App.scss'
import { Route, Switch, Redirect } from 'react-router-dom'
import Login from '../Login/Login'
import Signup from '../../components/Signup/Signup'
import { PropTypes } from 'prop-types'


class App extends Component {
  constructor() {
    super()
  }

  async componentDidMount() {
    const movies = await fetchData('https://api.themoviedb.org/3/movie/popular?api_key=da90047b6c1d3526d4b04666a1b64a0d&language=en-US&page=1&region=US')
    this.props.loadMovies(movies)
  }

  render() {
    const { user } = this.props
    return (
      <div className="App">
        <Switch>
          <Route 
            exact
            path='/'
            render={() => {
              if (!user.name) {
                return <Redirect to='/login' />
              } else {
                return <MovieDisplay />
              }
            }}
          />
          <Route
            path='/login'
            component={Login}
          />
          <Route
            path='/signup'
            component={Signup}
          />
        </Switch>
      </div>

    )
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  loadMovies: (movies) => dispatch(loadMovies(movies)),
})

App.propTypes = {
  movies: PropTypes.array.isRequired,
  user: PropTypes.object.isRequired,
  loadMovies: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(App)

