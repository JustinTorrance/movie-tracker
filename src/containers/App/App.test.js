import { mapStateToProps, mapDispatchToProps } from './App'
import { loadMovies } from '../../actions/index'
import { shallow } from 'enzyme'
import App from './App'
import React from 'react'

describe('App', () => {

  describe('App component', () => {

    it('Should match the snapshot', () => {
      const wrapper = shallow(<App />)
      expect(wrapper).toMatchSnapshot()
    })

    describe('ComponentDidMount', () => {
      
    })

  })

  describe('mapStateToProps', () => {
    it('should return an array with movie objects and a user object', () => {
      const mockMovie = {
        id: 1,
        title: 'Titanic',
        year: 1999,
        rating: 9,
        posterPic: 'https://image.tmdb.org/t/p/w500///',
        backdropPic: 'https://image.tmdb.org/t/p/w500///',
        overview: 'great movie',
        genres: 'drama',
        runtime: 50
      }

      const mockUser = {
        name: 'Ashley',
        email: 'Ashley@gmail.com',
        password: 'ashley'
      }
    
      const mockState = {
        movies: [mockMovie],
        user: mockUser
      }

      const expected = {
        movies: [mockMovie],
        user: mockUser
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

  })
  
  describe('mapDispatchToProps', () => {


    it('calls dispatch with a loadMovies action', () => {
      const mockMovie = {
        id: 1,
        title: 'Titanic',
        year: 1999,
        rating: 9,
        posterPic: 'https://image.tmdb.org/t/p/w500///',
        backdropPic: 'https://image.tmdb.org/t/p/w500///',
        overview: 'great movie',
        genres: 'drama',
        runtime: 50
      }
      const mockDispatch = jest.fn()
      const actionToDispatch = loadMovies(mockMovie)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.loadMovies(mockMovie)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)

    })
  })

})
