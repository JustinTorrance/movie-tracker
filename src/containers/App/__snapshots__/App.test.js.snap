// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    loadMovies={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://api.themoviedb.org/3/movie/popular?api_key=da90047b6c1d3526d4b04666a1b64a0d&language=en-US&page=1&region=US",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    movies={Array []}
    user={
      Object {
        "email": "jake",
        "name": "jake",
        "password": "jake",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          component={[MockFunction]}
          path="/login"
        />
        <Route
          component={[MockFunction]}
          path="/signup"
        />
        <Route
          component={[MockFunction]}
          path="/favorites"
        />
      </Switch>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            component={[MockFunction]}
            path="/login"
          />,
          <Route
            component={[MockFunction]}
            path="/signup"
          />,
          <Route
            component={[MockFunction]}
            path="/favorites"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            component={[MockFunction]}
            path="/login"
          />
          <Route
            component={[MockFunction]}
            path="/signup"
          />
          <Route
            component={[MockFunction]}
            path="/favorites"
          />
        </Switch>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              component={[MockFunction]}
              path="/login"
            />,
            <Route
              component={[MockFunction]}
              path="/signup"
            />,
            <Route
              component={[MockFunction]}
              path="/favorites"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [MockFunction],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [MockFunction],
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [MockFunction],
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
